## Adel Khorramrouz
### ak8480@rit.edu

In [1]:
from scipy.stats import lognorm, norm, expon
import numpy as np

### Q1
#### Let 𝑋 denote the amount of time for which a book on 2-hour reserve at a college library is checked out by a randomly selected student and suppose that 𝑋 has density function: 𝑓(𝑥) = {0.5𝑥 0 ≤ 𝑥 ≤ 2, 0 𝑜𝑡ℎ𝑒𝑟𝑤𝑖𝑠𝑒} Calculate the following probabilities:

#### a. 𝑃(𝑋 ≤ 1)

$$
P(X <= B) = CDF(X) = F(X) = \int_0^B \frac{x}{2} \, dx = \frac{x^2}{4}
\\
P(X <= 1) = F(1) =  \frac{1}{4}  = .25
$$

#### b. 𝑃(0.5 ≤ 𝑋 ≤ 1.5)

$$
F(1.5) - F(.5) = \frac{1.5^2}{4} - \frac{.5^2}{4} = \frac{2.25 - .25}{4} = .5
$$

#### c. 𝑃(𝑋 > 1.5)

$$
1 - F(1.5) = 1 - \frac{1.5^2}{4} = \frac{4 - 2.25}{4} = \frac{1.75}{4} = .4375
$$

### Q2
#### Extensive experience states that the time to failure of fans in certain types of diesel engines follow an exponential distribution with mean time to failure 10,000 hours.
#### a. What is the probability that the fan will last at least 15,000 hours

In [2]:
mean = 10000
exp_dist = expon(scale = mean)
print(1-exp_dist.cdf(15000))

0.2231301601484298


$$
X = time \ to\ failure
\\
distribution = exponential 
\\
\lambda = \frac{1}{mean} = \frac{1}{10000}
\\
P(X > 15,000) = 1 - F(15000) = .2231
$$

#### b. What is the standard deviation of time to failure?

$$
in \ exponential\  distribution\  we\  have:
\\
mean = \frac{1}{\lambda} = \sigma
\\
std = \sigma = 10000
$$

In [3]:
exp_dist.std()

10000.0

#### c. What is the probability that a fan will last longer than the mean by more than 2 standard deviations?

$$
P(X > \mu + 2*\sigma) = P (X > 30000) = 1 - F(30,000) = .04979
$$

In [4]:
print(1 - exp_dist.cdf(mean+mean+mean))

0.04978706836786395


### Q3
#### A person can be in an upright position either sitting or standing. For children between the ages of eight and fifteen, time per day in an upright position follows a normal probability model with mean 6.2 hours and standard deviation 1.4 hours.
#### a. What is the probability that a randomly selected child’s time per day in an upright position is more than 7 hours?


$$
X = time\ per\ day\ in\ an\ upright\ position
\\
Distribution = Normal
\\
\mu = 6.2
\\
\sigma = 1.4
\\
P (X > 7) = 1 - F(7) = .28385
$$

In [5]:
mean = 6.2
std = 1.4
normal_dist = norm(mean, std)
print(1 - normal_dist.cdf(7))

0.2838545830986763


#### b. What is the probability that a randomly selected child’s time per day in an upright position is less than 3 hours?

$$
P (X < 3) = F(3) = .01114
$$

In [6]:
print(normal_dist.cdf(3))

0.01113548947961638


#### c. Fill in the blank: 5% of children have less than ____ hours per day in an upright position.

$$
P(X < A) = F(A) = .05
\\
A = 3.89
$$

In [7]:
norm.ppf(.05,mean,std)

3.8972049222679384

#### d. Children in the highest 1% for time spent upright were in this position the longest. Therefore, 1% of children have more than ___ hours per day in an upright position.

$$
P(X > A) = 1 - F(A) = .01
\\
F(A) = .99
\\
A = 9.4569
$$

In [8]:
norm.ppf(.99,mean,std)

9.456887023657178

### Q4
#### The life of an Infinity brand car battery is normally distributed with mean 810 days and standard deviation 42 days.
#### a. What is the probability that a randomly selected Infinity car battery has a lifetime between 825 and 900 days?

$$
X = Battery\ lifetime
\\
Distribution = Normal
\\
\mu = 810
\\
\sigma = 42
\\
P(825 < X <900) = F(900) - F(825) = .3444
$$

In [9]:
mu = 810
std = 42
normal_dist = norm(mu, std)
print(normal_dist.cdf(900) - normal_dist.cdf(825))

0.3444301453470071


#### b. What is the probability that a randomly selected Infinity car battery has a lifetime more than 850 days?

$$
P(X > 850) = 1 - F(850) = .1705
$$

In [10]:
print(1 - normal_dist.cdf(850))

0.17045190796406517


#### c. What is the 37th percentile of brand Infinity lifetimes?

$$
P(X < A) = F(A) = .37
\\
A = 796.06
$$

In [11]:
percentile = .37
percentile_value = norm.ppf(percentile, loc=mu, scale=std)
print(f"The 37th percentile of brand Infinity lifetimes is approximately {percentile_value:.2f} days.")

The 37th percentile of brand Infinity lifetimes is approximately 796.06 days.


#### d. One Infinity battery had a lifetime of 700 days. Is this unusual? EXPLAIN.

<span style="color:blue">
in this question we have
$$
\frac{\left|{700-\mu}\right|}{\sigma} = \frac{810-700}{42} = 2.6
$$
Which means X is more than 2.6 times of standard deviation far from the mean. which makes it very unlikely and unusual 
the probablity of such an event is:
    
$$
P(X < 700) = F(700) = .0044
$$
    
Which is less than half percent
<span>

In [12]:
normal_dist.cdf(700)

0.004408781892886983

### Q5
#### Suppose that 𝑋 has a lognormal distribution with parameters 𝜇 = 6 and 𝜎 = 0.3. Determine the following:
#### a. 𝑃(𝑋 < 300)

In [13]:
mu = 6  # Mean of the underlying normal distribution
sigma = .3  # Standard deviation of the underlying normal distribution
scale = np.exp(mu)  # Scale parameter (corresponds to the median)
x = 300
lognorm_dist = lognorm(scale=scale, s=sigma)
print(lognorm_dist.cdf(300))

0.1617253133244424


#### b. The value for 𝑥 such that 𝑃(𝑋 ≤ 𝑥) = 0.95

In [14]:
lognorm_dist.ppf(.95)

660.8032153804501

#### c. 𝑃(100 < 𝑋 < 500)

In [15]:
lognorm_dist.cdf(500)-lognorm_dist.cdf(100)

0.7628051258906872